File: calculaPreco.ts
// Define uma interface para representar um produto
interface Produto {
  nome: string;
  preco: number;
  desconto?: number; // opcional
}

// Função para calcular o preço final com desconto
function calcularPrecoFinal(produto: Produto): number {
  const desconto = produto.desconto ?? 0; // usa 0 se não houver desconto
  return produto.preco - (produto.preco * desconto / 100);
}

// Exemplo de uso
const produto1: Produto = {
  nome: "Fone de ouvido",
  preco: 200,
  desconto: 15
};

console.log(`Preço final de ${produto1.nome}: R$ ${calcularPrecoFinal(produto1).toFixed(2)}`);

- Execute e observe o resultado no console.

- Altere o valor do desconto para testar o comportamento com novo percentual.

- Inclua linha de comando para impressão do "Preço inicial" com 2 casas decimais, acima da mensagem do "Preço final".

File: calculaPreco.py
# Define função para calcuclar preço final
def calcular_preco_final(nome, preco, desconto=0):
    preco_final = preco - (preco * desconto / 100)
    print(f"Preço final de {nome}: R$ {preco_final:.2f}")

# Exemplo de uso
calcular_preco_final("Fone de ouvido", 200, 15)
using System;

class Produto

{

    public string Nome { get; set; }

    public double Preco { get; set; }

    public double Desconto { get; set; }


    public double CalcularPrecoFinal()

    {

        return Preco - (Preco * Desconto / 100);

    }

}

class Program

{

    static void Main()

    {

        Produto produto = new Produto

        {

            Nome = "Fone de ouvido",

            Preco = 200,

            Desconto = 25

        };
        Console.WriteLine($"Preço iniical de {produto.Nome}: R$ {produto.Preco:F2}");
        Console.WriteLine($"Preço final de {produto.Nome}: R$ {produto.CalcularPrecoFinal():F2}");

    }

}
